src/v/storage/types.h

```diff
@@ -338,6 +338,15 @@ struct log_reader_config {
     friend std::ostream& operator<<(std::ostream& o, const log_reader_config&);
 };
 
+struct gc_config {
+    // remove everything below eviction time
+    model::timestamp eviction_time;
+    // remove one segment if log is > max_bytes
```

Comment:
nit: at some point we should rename this to `retained_bytes` or somesuch. With the new policy, `max_bytes` isn't quite accurate because we gladly keep more than this value if removal of a segment puts us below the value